src/main/java/edu/harvard/iq/dataverse/search/SolrSearchResult.java

```diff
@@ -401,6 +407,10 @@ public JsonArrayBuilder getRelevance() {
     }
 
     public JsonObject toJsonObject(boolean showRelevance, boolean showEntityIds, boolean showApiUrls) {
+        return toJsonObject(showRelevance, showEntityIds, showApiUrls, null);
+    }
+    
+    public JsonObject toJsonObject(boolean showRelevance, boolean showEntityIds, boolean showApiUrls, List<String> metadataFields) {
         return json(showRelevance, showEntityIds, showApiUrls).build();
     }
```

Comment:
This isn't working because you're passing the metadataFields to the toJsonObject method, but not to the json method at line 414

Reply:
You are right : I forget this part of code when I report my code from a previous git repository.
The correction is done